### PR TITLE
File abstractions for project config and env

### PR DESCRIPTION
**Pull request by tcdent**
Adds interfaces for reading/writing `agentstack.json` and `.env` files; `ConfigFile` and `EnvFile`, respectively. 

Ensures we're being responsible with user data and makes testing easy, too!

### CODE REVIEW

**Comment by tcdent** [comment]
Relies on the work I did yesterday, so once that gets merged this will be a much smaller changeset.

File: agentstack/generation/tool_generation.py [hunk 1]
```diff
@@ -87,9 +88,9 @@ def add_tool(tool_name: str, path: Optional[str] = None):
         path = './'
     
     framework = get_framework(path)
-    agentstack_json = open_json_file(f'{path}{AGENTSTACK_JSON_FILENAME}')
+    agentstack_config = ConfigFile(path)
```
Comment (bboynton97) [inline agentstack/generation/tool_generation.py:91]: 🧼🧼🧼

**Comment by bboynton97** [comment]
this is the most beautiful thing i didnt even know i wanted

**Review by braelyn-ai — APPROVED** [review]

**Comment by potpiedev** [comment]
## Starting Review

This PR ([Vanta] Fix Low vulnerabilities in potpie-workflows) addresses security vulnerabilities identified in the potpie-workflows repository. The review is now underway and should be available shortly.

**Pull request merged**